Adds nonexistent status informer rule and alphabetizes (#855)
replicatedhq/replicated-docs: docs/reference/linter.mdx

    <td>
      Files with <code>kind: Application</code> and <code>apiVersion: kots.io/v1beta1</code>.
    </td>
  </tr>
  <tr>
    <th>Example</th>
    <td><p>Example of correct YAML for this rule:</p><ApplicationStatusInformers/></td>
  </tr>
</table>

### config-option-invalid-type

<table>
  <tr>
    <th>Description</th>
    <td>
      <p>Enforces valid types for Config items.</p>
      <p>For more information, see <a href="/reference/custom-resource-config#items">Items</a> in <em>Config</em>.</p>
    </td>
  </tr>
  <tr>
    <th>Level</th>
    <td>Error</td>
  </tr>
  <tr>
    <th>Applies To</th>
    <td>All files</td>
  </tr>
  <tr>
    <th>Example</th>
    <td><ConfigOptionInvalidType/></td>
  </tr>
</table>

### config-option-is-circular

<table>
  <tr>
    <th>Description</th>
    <td>
      Enforces that all ConfigOption items do not reference themselves.
  </td>
  </tr>
  <tr>
    <th>Level</th>
    <td>Error</td>
  </tr>
  <tr>
    <th>Applies To</th>
    <td>
      Files with <code>kind: Config</code> and <code>apiVersion: kots.io/v1beta1</code>.
    </td>
  </tr>
  <tr>
    <th>Example</th>
    <td> <ConfigOptionIsCircular/> </td>
  </tr>
</table>

> ### config-option-password-type

Reviewer: I think this rule should follow "config-option-not-found" and "config-option-not-repeatable".

    <td><p>Example of matching YAML for this rule:</p><MayContainSecrets/></td>
  </tr>
</table>

### missing-api-version-field

<table>
  <tr>
    <th>Description</th>
    <td>Requires the <code>apiVersion:</code> field in all files.</td>
  </tr>
  <tr>
    <th>Level</th>
    <td>Error</td>
  </tr>
  <tr>
    <th>Applies To</th>
    <td>All files</td>
  </tr>
  <tr>
    <th>Example</th>
    <td><p>Example of correct YAML for this rule:</p>
    <MissingAPIVersionField/>
    </td>
  </tr>
</table>

### missing-kind-field

<table>
  <tr>
    <th>Description</th>
    <td>Requires the <code>kind:</code> field in all files.</td>
  </tr>
  <tr>
    <th>Level</th>
    <td>Error</td>
  </tr>
  <tr>
    <th>Applies To</th>
    <td>All files</td>
  </tr>
  <tr>
    <th>Example</th>
    <td><p>Example of correct YAML for this rule:</p>
    <MissingKindField/>
    </td>
  </tr>
</table>

### nonexsistent-status-informer-object

<table>
  <tr>
    <th>Description</th>
    <td>Requires that the value of the <code>statusInformers</code> field is an existing Kubernetes object.</td>
  </tr>
  <tr>
    <th>Level</th>
    <td>Warning</td>
  </tr>
  <tr>
    <th>Applies To</th>
    <td>

>       <p>All files.</p>

Reviewer 2: ```suggestion
      <p>This will detect `statusInformer` entries in a `kots.io/v1beta1` `Application` manifest, but compares those to all manifests in the release.</p>
```

>       <p>Does not apply to Helm resources. Ignore any <code>nonexsistent-status-informer-object</code> warnings that appear for a Helm resource status informer.</p>

Reviewer 2: ```suggestion
      <p>Because the linter does not template helm charts during static analysis, it cannot evaulate <code>statusInformer</code> entries for helm-managed resources. If you have configured status informers for your helm workloads, you can ignore <code>nonexistent-status-informer-object</code> warnings for those workloads. If you find these warnings to be noisy, you can use a <a href="https://docs.replicated.com/reference/custom-resource-lintconfig">LintConfig</a> manifest to set them to `info` or `off`.</p>
```

  </tr>
  <tr>
    <th>Example</th>
    <td><p>Example of matching YAML for this rule:</p><MayContainSecrets/></td>
  </tr>
</table>

### missing-api-version-field

<table>
  <tr>
    <th>Description</th>
    <td>Requires the <code>apiVersion:</code> field in all files.</td>
  </tr>
  <tr>
    <th>Level</th>
    <td>Error</td>
  </tr>
  <tr>
    <th>Applies To</th>
    <td>All files</td>
  </tr>
  <tr>
    <th>Example</th>
    <td><p>Example of correct YAML for this rule:</p>
    <MissingAPIVersionField/>
    </td>
  </tr>
</table>

### missing-kind-field

<table>
  <tr>
    <th>Description</th>
    <td>Requires the <code>kind:</code> field in all files.</td>
  </tr>
  <tr>
    <th>Level</th>
    <td>Error</td>
  </tr>
  <tr>
    <th>Applies To</th>
    <td>All files</td>
  </tr>
  <tr>
    <th>Example</th>
    <td><p>Example of correct YAML for this rule:</p>
    <MissingKindField/>
    </td>
  </tr>
</table>

### nonexistent-status-informer-object

<table>
  <tr>
    <th>Description</th>
    <td>

>       <p>Requires that the value of the <code>statusInformers</code> field is an existing Kubernetes object.</p>

Reviewer 2: want to clarify this, mostly replace `is` with `references`, but open to further iteration from here

```suggestion
      <p>Requires that the each <code>statusInformers</code> entry references an existing Kubernetes workload.</p>
```
